src/robot/libraries/String.py

```diff
@@ -121,6 +123,45 @@ def decode_bytes_to_string(self, bytes, encoding, errors='strict'):
             raise TypeError('Can not decode strings on Python 3.')
         return bytes.decode(encoding, errors)
 
+    def format_string(self, template, *positional, **named):
+        """Formats a ``string`` using the given ``positional`` and ``named``
+        arguments.
+
+        If the given ``string`` is a valid absolute file path, opens the file in read
+        mode and then format its content using the given
+        ``positional`` and ``named`` arguments. Trailing new lines are removed
+        before formating the string. The spaces are kept.
+
+        This keyword uses python's string format. For more information see:
+        [https://docs.python.org/library/string.html#formatstrings]|Format syntax]
+
+        Examples:
+
+        `Considering the file C:\\template.txt contents as "My {test} String"`
+        | ${result} = | Format String | C:\\template.txt | | test=awesome |
+        | Should Be Equal | ${result} | My awesome String |
+        | ${result} = | Format String | User {} is not a admin user. | non-admin | |
+        | Should Be Equal | ${result} | User non-admin is not a admin user. |
+        | ${result} = | Format String | Username: {username} - Password: {password} | | username=Robot | password=Framework |
+        | Should Be Equal | ${result} | Username: Robot - Password: Framework |
+        | ${result} = | Format String | Document {} is missing on folder {folder} | tests.robot | folder=/home |
+        | Should Be Equal | ${result} | Document tests.robot is missing on folder /home |
+        | ${result} = | Format String | Uploaded file: {} should not be bigger than {}. | photo.jpg | 5MB |
+        | Should Be Equal | ${result} | Uploaded file: photo.jpg should not be bigger than 5MB. |
+
+        New in Robot Framework 3.1.
+        """
+        if os.path.isabs(template) and os.path.isfile(template):
+            template = template.replace("/", os.sep)
```

Comment:
Why is this replacement needed? The check in the previous like has already deemed the value as an existing file for the current OS,  replacing its forwarded slashes now is a bit late (and changes the filename).  
By the replacement I guess this is for Windows clients to be able to use *nix path values, but Windows does work with both path separators (`\` and `/`).

Reply:
The main reason is that logging is consistent. I doubt that's a big issue for users, but it makes it easier to create tests. Not sure is the log message actually tested, though.